internal/transport/keepalive_test.go

```diff
@@ -326,8 +300,8 @@ func (s) TestKeepaliveClientClosesWithActiveStreams(t *testing.T) {
 	copts := ConnectOptions{
 		ChannelzParentID: channelz.NewIdentifierForTesting(channelz.RefSubChannel, time.Now().Unix(), nil),
 		KeepaliveParams: keepalive.ClientParameters{
-			Time:    1 * time.Second,
-			Timeout: 1 * time.Second,
+			Time:    500 * time.Millisecond,
```

Comment:
This races with the creation of the first stream, which is why you had to use a "big" value like 500ms.

How hard would it be to make the "no ping server" actually respond to pings until you set a flag, at which point it stops responding?  Then we could create the stream, then set the flag, and the values here could be much lower.

Reply:
I checked. it seems doable. but i would like to pick this up later. I've filed an [issue](https://github.com/grpc/grpc-go/issues/6099) and added a TODO. lmk if this works